Update release version to 0.5.0

aarondav opened the pull request.

aarondav committed:
Update release version to 0.5.0

aarondav committed:
Update CHANGELOG.rst

aarondav committed:
More updates

aarondav committed:
Add summary

aarondav committed:
Add two more commits

aarondav committed:
Add forthcoming commit

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,52 @@
+    (Note that in a future version, we will likely rename or alias ``run_uuid``.)
+
+  - The Python entities returned by the servive API are now accessible in ``mlflow.entities`` directly. Where previously you may have used ``mlflow.entities.experiment.Experiment``, you would now just use ``mlflow.entities.Experiemnt``. The previous version still exists, but is deprecated and may be hidden in a future version.
+- url-parse dependency updated to 1.4.3 due to CVE in previous versions (#292, @mateiz)
```
This is not a breaking change; it's in JS

aarondav committed:
Squashed commit of the following:

commit 58626fb3229290f4955063efb802b7c2276c7c52
Author: Aaron Davidson <aaron@databricks.com>
Date:   Fri Aug 17 14:45:01 2018 -0700

    Update rst

commit c863edf61ff545521d6dfe32d44117e721d69e7d
Author: Aaron Davidson <aaron@databricks.com>
Date:   Fri Aug 17 09:31:16 2018 -0700

    Add docs for mlflow.tracking

smurching commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,52 @@
+
+    (Note that in a future version, we will likely rename or alias ``run_uuid``.)
+
+  - The Python entities returned by the servive API are now accessible in ``mlflow.entities`` directly. Where previously you may have used ``mlflow.entities.experiment.Experiment``, you would now just use ``mlflow.entities.Experiemnt``. The previous version still exists, but is deprecated and may be hidden in a future version.
```
servive -> REST?

aarondav committed:
Updates

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,51 @@
+
+- The Tracking API has been split into two pieces, a "basic logging" and a "tracking service" API. The "basic logging" API deals with logging metrics, parameters, and artifacts to the currently-active active run, and is accessible in ``mlflow`` (e.g., ``mlflow.log_param``). The tracking service API allow managing experiments and runs (especially historical runs) and is available in ``mlflow.track ...
+
+  - ``mlflow.tracking`` no longer exposes the fluent API, only ``mlflow``. So, code that was written like ``from mlflow.tracking import log_param`` will have to be ``from mlflow import log_param`` (note that almost all examples were already doing this).
```
Do we still want to use the word fluent here?

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,51 @@
+
+Breaking changes:
+
+- The Tracking API has been split into two pieces, a "basic logging" and a "tracking service" API. The "basic logging" API deals with logging metrics, parameters, and artifacts to the currently-active active run, and is accessible in ``mlflow`` (e.g., ``mlflow.log_param``). The tracking service API allow managing experiments and runs (especially historical runs) and is available in ``mlflow.track ...
```
`a "basic logging" and a "tracking service" API` -> `a "basic logging" API and a "tracking service" API`

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,51 @@
+
+  - ``mlflow.tracking`` no longer exposes the fluent API, only ``mlflow``. So, code that was written like ``from mlflow.tracking import log_param`` will have to be ``from mlflow import log_param`` (note that almost all examples were already doing this).
+  - Access to the service API goes through the ``mlflow.tracking.get_service()`` function, which relies on the same tracking server set by either the environment variable ``MLFLOW_TRACKING_URI`` or by code with ``mlflow.tracking.set_tracking_uri()``. So code that used to look like ``mlflow.tracking.get_run()`` will now have to do ``mlflow.tracking.get_service().get_run()``. This does not apply to ...
+  - ``mlflow.ActiveRun`` has been converted into a lightweight wrapper around ``mlflow.entities.Run`` to enable the Python ``with`` syntax. This means that there are no longer any special methods on the object` returned when calling ``mlflow.start_run()``. These can be converted to the service API. For example, ``mlflow.start_run().set_terminated()`` would now be called as:
```
Extra \` after `object`

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,51 @@
+        run = mlflow.start_run()
+        mlflow.tracking.get_service().set_terminated(run.info.run_uuid)
+
+    (Note that in a future version, we will likely rename or alias ``run_uuid``.)
```
Not sure that we need the example or the note here. I don't think people were calling methods on the returned objects super often.

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,51 @@
+
+Bug fixes:
+
+- Fix numpy array array serialization for int64 and other related types, allowing pyfunc to return such results (#240, @arinto)
```
array array

aarondav committed:
Address comments

aarondav committed:
servive

mparkhe commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,43 @@
+  - Access to the service API goes through the ``mlflow.tracking.get_service()`` function, which relies on the same tracking server set by either the environment variable ``MLFLOW_TRACKING_URI`` or by code with ``mlflow.tracking.set_tracking_uri()``. So code that used to look like ``mlflow.tracking.get_run()`` will now have to do ``mlflow.tracking.get_service().get_run()``. This does not apply to ...
+  - ``mlflow.ActiveRun`` has been converted into a lightweight wrapper around ``mlflow.entities.Run`` to enable the Python ``with`` syntax. This means that there are no longer any special methods on the object returned when calling ``mlflow.start_run()``. These can be converted to the service API.
+
+  - The Python entities returned by the servive API are now accessible in ``mlflow.entities`` directly. Where previously you may have used ``mlflow.entities.experiment.Experiment``, you would now just use ``mlflow.entities.Experiemnt``. The previous version still exists, but is deprecated and may be hidden in a future version.
```
Experiment typo

aarondav committed:
Add new commit

mparkhe commented on docs/source/tracking.rst:
```diff
@@ -156,6 +158,21 @@ environment variable.
         mlflow.log_param("a", 1)
```
documentation for `log_param` in `fluent` says that value should be a `string` (or maybe we should document that param value gets cast to string in service)

aarondav committed:
Install torch via pip instead of whl

aarondav committed:
Fix documentation markup

aarondav committed:
Address Mani's comments

aarondav commented:
@mparkhe thanks, comments addressed!

While building the docs, I noticed that our test-requirements for PyTorch seemed to be pulling a Linux-only wheel, so I changed it to pip -- let's see if tests pass.

I also fixed up a bunch of references in the docs that I noticed were messed up.
